### PR TITLE
Set configurable keepalive socket option for the audit server and sudoers

### PR DESCRIPTION
**Pull request by laczau**

### CODE REVIEW

File: logsrvd/logsrvd.c [hunk 1]
```diff
@@ -1503,6 +1504,17 @@ listener_cb(int fd, int what, void *v)
 	}
 	/* TODO: pause accepting on ENFILE and EMFILE */
     }
+
+    /* set keepalive socket option on socket returned by accept */
+    if (logsrvd_conf_tcp_keepalive()) {
+        int keepalive = 1;
+        if (setsockopt(sock, SOL_SOCKET, SO_KEEPALIVE, &keepalive,
+            sizeof(keepalive)) < 0) {
```
Comment (millert) [inline logsrvd/logsrvd.c]: Please compare to -1 instead of <0 here.

File: plugins/sudoers/iolog_client.c [hunk 2]
```diff
@@ -166,6 +166,20 @@ connect_server(const char *host, const char *port, struct timespec *timo,
 	    sock = -1;
 	    continue;
 	}
+
+    if (tcp_keepalive) {
+        int keepalive = 1;
+        if (setsockopt(sock, SOL_SOCKET, SO_KEEPALIVE, &keepalive,
```
Comment (millert) [inline plugins/sudoers/iolog_client.c]: Please compare to -1 instead of <0 here.

File: plugins/sudoers/def_data.in [hunk 3]
```diff
@@ -372,6 +372,9 @@ log_servers
 log_server_timeout
 	T_TIMEOUT|T_BOOL
 	"Sudo log server timeout in seconds: %u"
+log_server_disable_keepalive
```
Comment (millert) [inline plugins/sudoers/def_data.in]: I would really prefer to have this be log_server_keepalive and initialize it to true in init_defaults().  If someone doesn't want it enabled they can negate the option in sudoers.  That way the logic is more consistent between the client and server code.

File: logsrvd/logsrvd.c (hunk 1 above)
Comment (laczau) [reply logsrvd/logsrvd.c]: fixed

File: plugins/sudoers/iolog_client.c (hunk 2 above)
Comment (laczau) [reply plugins/sudoers/iolog_client.c]: fixed

File: plugins/sudoers/def_data.in (hunk 3 above)
Comment (laczau) [reply plugins/sudoers/def_data.in]: fixed, the logic has been negated, log_server_disable_keepalive has been changed to log_server_keepalive

**Pull request merged**